blockchain/state_migration.go

```diff
@@ -144,32 +148,29 @@ func (bc *BlockChain) migrateState(rootHash common.Hash) error {
 			}
 			results[i] = result
 		}
-		read, readElapsed := len(queue), time.Since(startIter)
+		stats.read += len(queue)
+		stats.readElapsed += time.Since(startRead)
 
 		// Process trie nodes
+		startProcess := time.Now()
 		if _, index, err := trieSync.Process(results); err != nil {
 			logger.Error("State migration is failed by process error", "err", err)
 			return fmt.Errorf("failed to process result #%d: %v", index, err)
 		}
+		stats.processElapsed += time.Since(startProcess)
 
 		// Commit trie nodes
-		written, writeElapsed, err := bc.stateMigrationCommit(trieSync, targetDB.DiskDB())
+		startWrite := time.Now()
+		written, err := bc.stateMigrationCommit(trieSync, stateTrieBatch)
 		if err != nil {
 			logger.Error("State migration is failed by commit error", "err", err)
 			return fmt.Errorf("failed to commit data #%d: %v", written, err)
 		}
+		stats.committed += written
+		stats.writeElapsed += time.Since(startWrite)
 
 		// Report progress
-		committedCnt += written
-		bc.committedCnt, bc.pendingCnt, bc.progress = committedCnt, trieSync.Pending(), trieSync.CalcProgressPercentage()
-		progressStr := strconv.FormatFloat(bc.progress, 'f', 4, 64)
-		progressStr = strings.TrimRight(progressStr, "0")
-		progressStr = strings.TrimRight(progressStr, ".") + "%"
-
-		logger.Warn("State migration progress",
-			"progress", progressStr, "committedCnt", committedCnt, "pendingCnt", bc.pendingCnt,
-			"read", read, "readElapsed", readElapsed, "written", written, "writeElapsed", writeElapsed,
-			"elapsed", time.Since(startIter))
+		stats.stateMigrationReport(false, trieSync.Pending(), trieSync.CalcProgressPercentage())
```

Comment:
We need to change `CalcProgressPercentage()` if the size of `trieSync.Missing` is changed.